keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/PersistentEvent.kt

```diff
@@ -34,3 +34,24 @@ abstract class PersistentEvent {
     @JsonProperty("application") APPLICATION
   }
 }
+
+/**
+ * Common interface implemented by all [ResourceEvent]s and certain [ApplicationEvent]s that affect all the
+ * application's resources, such as pausing and resuming.
+ */
+@JsonTypeInfo(
+  use = JsonTypeInfo.Id.NAME,
+  property = "type",
+  include = JsonTypeInfo.As.PROPERTY
+)
+@JsonSubTypes(
+  JsonSubTypes.Type(value = ResourceEvent::class),
+  JsonSubTypes.Type(value = ApplicationActuationPaused::class),
+  JsonSubTypes.Type(value = ApplicationActuationResumed::class)
+)
+interface ResourceHistoryEvent {
+  val scope: PersistentEvent.Scope
+  val uid: String // the resource ID or application name
```

Comment:
is `uid` the right name for this variable then? what about `identifier`?

Reply:
It's really not. I didn't change it here because @robfletcher renamed it to `ref` in #1151.